spec/unit/dry/configurable/dsl_spec.rb

```diff
@@ -43,10 +43,21 @@
   end
 
   it 'compiles a setting with a constructor' do
+    setting = dsl.setting(:dsn, 'sqlite', constructor: ->(value) { "jdbc:#{value}" })
+
+    expect(setting.name).to be(:dsn)
+    expect(setting.value).to eql('jdbc:sqlite')
+  end
+
+  it 'supports but deprecates giving a constructor as a block' do
+    logger = StringIO.new
+    Dry::Core::Deprecations.logger(logger)
```

Comment:
@waiting-for-dev you can move this to `spec_helper` and use `Logger.new("log/deprecations.log")` there

Reply:
Unless I'm missing something, this will make the logger persistent across tests, so we'll be sharing a state between them. I think it's not ideal unless we can't avoid it. What do you think creating about creating a helper method so that we can reuse it easily? For instance:

```ruby
with_logger do |log|
  # expectation
  expect(log).to include('....')
end
```

Reply:
@waiting-for-dev OK let's leave it here since it's the only place where we need it for now. Eventually we should have something reusable though, that can be provided by dry-core.

Reply:
OK, it's going to be very soon, as I'm going to use it again in the PR that will deprecate providing the default as positional argument :smile: